banyand/measure/merger_algo.go

```diff
@@ -0,0 +1,118 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package measure
+
+import (
+	"sort"
+)
+
+var (
+	_ MergePolicy = (*noopMergePolicy)(nil)
+	_ MergePolicy = (*lowestWriteAmplificationMergePolicy)(nil)
+)
+
+// MergePolicy is to select parts to be merged.
+type MergePolicy interface {
+	GetPartsToMerge(src []*partWrapper, maxFanOut uint64) []*partWrapper
+}
+
+// noopMergePolicy aims to merge all parts.
+type noopMergePolicy struct{}
+
+func (s *noopMergePolicy) GetPartsToMerge(src []*partWrapper, _ uint64) []*partWrapper {
+	return src
+}
+
+// lowestWriteAmplificationMergePolicy aims to choose an optimal combination
+// that has the lowest write amplification.
+// This policy can be referred to https://github.com/VictoriaMetrics/VictoriaMetrics/blob/v0.4.2-victorialogs/lib/logstorage/datadb.go
+// But parameters can be varied.
+type lowestWriteAmplificationMergePolicy struct {
+	maxParts           int
+	minMergeMultiplier float64
```

Comment:
It seems a bit tricky to set it.

Reply:
I want the parameters to be adjusted both for testing and for users since due to our experience with VictoriaLogs, different workload/env need various parameter set.